pkg/process/metadata/parser/java/jee.go

```diff
@@ -0,0 +1,253 @@
+	wlsHomeSysProp      string = "-Dwls.home="
+	websphereJar        string = "ws-server.jar"
+	websphereMainClass  string = "defaultServer"
+	tomcatMainClass     string = "org.apache.catalina.startup.Bootstrap"
+	tomcatSysProp       string = "-Dcatalina.base="
+	jbossStandaloneMain string = "org.jboss.as.standalone"
+	jbossDomainMain     string = "org.jboss.as.server"
+	jbossSysProp        string = "-Djboss.home.dir="
+	applicationXMLPath  string = "/META-INF/application.xml"
+)
+
+var (
+	errUnhandledDeployment = errors.New("unhandled deployment type")
+)
+
+type (
+	// applicationXML is used to unmarshal information from a standard EAR's application.xml
+	// example doc: https://docs.oracle.com/cd/E13222_01/wls/docs61/programming/app_xml.html
+	applicationXML struct {
+		XMLName     xml.Name `xml:"application"`
+		ContextRoot []string `xml:"module>web>context-root"`
+	}
+	// deployedAppFindFn is used to find the application deployed on a domainHome
+	// args should be supplied since some vendors may require additional information from them (i.e. server name)
+	deployedAppFindFn func(domainHome string, args []string, fs afero.Fs) ([]string, bool)
+	// warContextRootFindFn is used to extract the context root from a vendor defined configuration inside the war.
+	// if not found it returns en empty string and false
+	warContextRootFindFn func(fs afero.Fs) (string, bool)
+	// defaultWarContextRootFn returns the default naming that apply for a certain fileName.
+	// it is usually the file without the extension, but it can differ for some vendors (i.e. tomcat)
+	defaultWarContextRootFn func(fileName string) string
+)
+
+// definitions of standard extractors
+var (
+	deploymentFinders = map[serverVendor]deployedAppFindFn{
+		weblogic: weblogicFindDeployedApps,
+	}
+	contextRootFinders = map[serverVendor]warContextRootFindFn{
+		weblogic: weblogicExtractWarContextRoot,
+	}
+	defaultContextNameExtractors = map[serverVendor]defaultWarContextRootFn{
+		weblogic: standardExtractContextFromWarName,
+	}
+)
+
+// extractContextRootFromApplicationXML parses a standard application.xml file extracting
+// mount points for web application (aka context roots).
+func extractContextRootFromApplicationXML(fs afero.Fs) ([]string, error) {
+	reader, err := fs.Open(applicationXMLPath)
+	if err != nil {
+		return nil, err
+	}
+	defer reader.Close()
+	var a applicationXML
+	err = xml.NewDecoder(reader).Decode(&a)
+	if err != nil {
+		return nil, err
+	}
+	return a.ContextRoot, nil
+}
+
+// resolveAppServerFromCmdLine parses the command line and tries to extract a couple of evidences for each known application server.
+// The first is the server home (usually defined by a system property) and the second is the main class / jar entry point.
+// It returns the serverVendor (unknown in case it cannot be determined with precision) and the domain path if applicable.
+func resolveAppServerFromCmdLine(args []string) (serverVendor, string) {
+	hint1, hint2 := unknown, unknown
```

Comment:
what does hint1 and hint2 suppose to mean?

Reply:
As I put on the comments, one hint is about matching the entrypoint (i.e. the class name or the jar being executed) another one is matching a characteristic property (i.e. the weblogic server home). The reasons why we need both hints to match is that, in some cases the same jar may be used for admin operations (not to launch the server) or the same property may be used for admin operation and not to launch the server (like happening for weblogic). It's a bit complex to enumerate all the possible cases but having a double verification helps to avoid false positives when detecting a server vendor. Then comes the needs of having a bitwise enum since, in this case we can just logic AND both hints having the vendor if matching both, otherwise 0 that's `unknown`

Reply:
I dont think the code is clear to the average reader
We can have better names than hint1, and hint2
we can have the long explanation inside the code to make it easier

Reply:
I added way more details and renamed the variables for a better understanding